flang/test/Lower/optional.f90

```diff
@@ -0,0 +1,122 @@
+! RUN: bbc -emit-fir %s -o - | FileCheck %s
+
+! Test OPTIONAL lowering on caller/callee. Also test PRESENT intrinsic.
+module opt
+contains
+
+! Test simple scalar optional
+! CHECK-LABEL: func @_QMoptPintrinsic_scalar(%arg0: !fir.ref<f32>)
+subroutine intrinsic_scalar(x)
+  implicit none
+  real, optional :: x
+  ! CHECK: %[[addr:.*]] = fir.convert %arg0 : (!fir.ref<f32>) -> index
```

Comment:
Should `%arg0` be turned into a variable match?

Reply:
So far I was skipping the complexity of the CHECK-LABEL + CHECK-SAME in lit tests to capture arguments, but I think you are right, its more future proof to capture it and it is easy here. I updated the tests.